nativelib/src/main/scala/scala/scalanative/native/pure.scala

```diff
@@ -0,0 +1,10 @@
+package scala.scalanative
+package native
+
+/** Let optimizer assume that given method is pure.
+  * This means that repetitive calls to the same method
+  * with the same arguments are allowed to be eliminated.
+  * Additionally, if method result to given method is not used
+  * then the call can be elided altogether.
+  */
+final class pure extends scala.annotation.StaticAnnotation
```

Comment:
[`@pure` was originally added to completely get rid off boxing around polymorphic builtins like casts and pointer operations.](https://github.com/scala-native/scala-native/commit/554be157aadddf3f71e9b06e6ad1f2cd06e73cf0). It used to allocate two boxes on every such operation at some time and that's really bad for a low-level primitive.